sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateOrdering.scala

```diff
@@ -71,7 +71,13 @@ object GenerateOrdering extends CodeGenerator[Seq[SortOrder], BaseOrdering] with
     ctx.INPUT_ROW = row
     // to use INPUT_ROW we must make sure currentVars is null
     ctx.currentVars = null
-    ordering.map(_.child.genCode(ctx))
+    ordering.map(_.child.genCode(ctx)) match {
+      case stream: Stream[ExprCode] =>
```

Comment:
This is not the first time that we are bitten by `Stream`, maybe we should use `IndexedSeq` instead of `Seq` in codegen APIs. cc @rednaxelafx @viirya @maropu @kiszk

Reply:
I agree. From a quick survey (`git grep -i -e map --and -e genCode`), a number of call sites seem vulnerable to the same bug.

Reply:
+1

Reply:
+1